openedx/adg/lms/applications/handlers.py

```diff
@@ -31,3 +34,30 @@ def delete_business_line_group(instance, **kwargs):  # pylint: disable=unused-ar
     """
     # TODO: Use instance.group.delete(), once LP-2479 is deployed on prod.
     Group.objects.filter(name=instance.title).delete()
+
+
+@receiver(post_save, sender=CourseEnrollment)
+def update_prereq_courses_status(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update prerequisite course groups status to IN_PROGRESS in ApplicationHub when user in enrolled in
+    prereq course.
+    """
+    if not instance.is_active:
+        return
+
+    hub, _ = ApplicationHub.objects.get_or_create(user=instance.user)
+    is_any_status_updated = False
+
+    is_program_pre_req = is_course_in_prereqs(instance.course)
+    if is_program_pre_req and hub.program_prereq_courses_status == ApplicationHub.NOT_STARTED:
+        hub.program_prereq_courses_status = ApplicationHub.IN_PROGRESS
+        is_any_status_updated = True
```

Comment:
The `is_course_in_prereqs` is a heavy function. So before executing that, we should first be checking the condition `hub.program_prereq_courses_status == ApplicationHub.NOT_STARTED`, so that if this condition is not met, we no longer need to call the function.

Something like this would do:

```
if hub.program_prereq_courses_status == ApplicationHub.NOT_STARTED:
   if is_course_in_prereqs(instance.course):
       hub.program_prereq_courses_status = ApplicationHub.IN_PROGRESS
       is_any_status_updated = True
```

We would be doing the same with the second condition below.